testframework/src/main/java/net/neoforged/testframework/summary/FormattingUtil.java

```diff
@@ -0,0 +1,193 @@
+            double minDistance = Double.MAX_VALUE;
+            int closestColorIndex1 = 0;
+
+            // Iterate over the ANSI colors and find the one with the minimum distance
+            for (int i = 0; i < COLORS_RGB.length; i++) {
+                int[] ansiColor = COLORS_RGB[i];
+                double distance = calculateDistance(red, green, blue, ansiColor[0], ansiColor[1], ansiColor[2]);
+                if (distance < minDistance) {
+                    minDistance = distance;
+                    closestColorIndex1 = i;
+                }
+            }
+            int closestColorIndex = closestColorIndex1;
+
+            if (closestColorIndex > 7) {
+                return "\u001B[" + (90 + (closestColorIndex - 8)) + "m";
+            }
+            return "\u001B[" + (30 + closestColorIndex) + "m";
+        }
+
+        private static double calculateDistance(int red1, int green1, int blue1, int red2, int green2, int blue2) {
+            // Convert CIE XYZ to CIE LAB
+            double[] lab1 = xyzToLab(red1, green1, blue1);
+            double[] lab2 = xyzToLab(red2, green2, blue2);
+
+            // Compute Euclidean distance in CIE LAB space
+            double deltaL = lab2[0] - lab1[0];
+            double deltaA = lab2[1] - lab1[1];
+            double deltaB = lab2[2] - lab1[2];
+            return Math.sqrt(deltaL * deltaL + deltaA * deltaA + deltaB * deltaB);
+        }
+
+        private static double[] xyzToLab(double r, double g, double b) {
+            r /= 255.0;  //R from 0 to 255
+            g /= 255.0;  //G from 0 to 255
+            b /= 255.0;  //B from 0 to 255
+
+            if (r > 0.04045) r = Math.pow(((r + 0.055) / 1.055), 2.4);
+            else r = r / 12.92;
+            if (g > 0.04045) g = Math.pow(((g + 0.055) / 1.055), 2.4);
+            else g = g / 12.92;
+            if (b > 0.04045) b = Math.pow(((b + 0.055) / 1.055), 2.4);
+            else b = b / 12.92;
+
+            r *= 100;
+            g *= 100;
+            b *= 100;
+
+            // Observer. = 2°, Illuminant = D65
+            double x = r * 0.4124 + g * 0.3576 + b * 0.1805;
+            double y = r * 0.2126 + g * 0.7152 + b * 0.0722;
+            double z = r * 0.0193 + g * 0.1192 + b * 0.9505;
+
+            x /= 95.047;         //ref_X =  95.047     Observer= 2°, Illuminant= D65
+            y /= 100.000;        //ref_Y = 100.000
+            z /= 108.883;        //ref_Z = 108.883
+
+            if (x > 0.008856) x = Math.pow(x, 1.0 / 3);
+            else x = (7.787 * x) + (16.0 / 116);
+            if (y > 0.008856) y = Math.pow(y, 1.0 / 3);
+            else y = (7.787 * y) + (16.0 / 116);
+            if (z > 0.008856) z = Math.pow(z, 1.0 / 3);
+            else z = (7.787 * z) + (16.0 / 116);
+
+            return new double[] {
+                    (116 * y) - 16, // CIE-L
+                    500 * (x - y),  // CIE-a
+                    200 * (y - z),  // CIE-b
+            };
+        }
+    }
```

Comment:
This smells like stackoverflow. If it is from a SO post or whatever, please link said post.

Reply:
nope that was chat gpt + research and tweaking

Reply:
Great, I hope it works because I don't feel like double-checking magic numbers.

Reply:
don't worry I tested it